packages/pluggableWidgets/column-chart-native/src/components/ColumnChart.tsx

```diff
@@ -0,0 +1,367 @@
+import { createElement, ReactElement, useCallback, useMemo, useState } from "react";
+import { LayoutChangeEvent, Text, TextStyle, View } from "react-native";
+import { VictoryAxis, VictoryBar, VictoryChart, VictoryGroup, VictoryStack } from "victory-native";
+import { BarProps } from "victory-bar";
+import { extractStyles } from "@mendix/pluggable-widgets-tools";
+
+import { ColumnChartStyle } from "../ui/Styles";
+import { SortOrderEnum } from "../../typings/ColumnChartProps";
+import { Legend } from "./Legend";
+import { aggregateGridPadding, mapToAxisStyle, mapToGridStyle, mapToColumnStyles } from "../utils/StyleUtils";
+
+export interface ColumnChartProps {
+    name: string;
+    series: ColumnChartSeries[];
+    sortOrder: SortOrderEnum;
+    style: ColumnChartStyle;
+    presentation: string;
+    showLegend: boolean;
+    showLabels: boolean;
+    xAxisLabel?: string;
+    yAxisLabel?: string;
+    warningPrefix?: string;
+}
+
+export interface ColumnChartSeries {
+    dataPoints: ColumnDataPoints;
+    xFormatter?: (xValue: number | Date | string) => string;
+    yFormatter?: (yValue: number | Date | string) => string;
+    name?: string;
+    customColumnStyle?: string;
+}
+
+export type ColumnDataPoints =
+    | Array<ColumnDataPoint<number, number>>
+    | Array<ColumnDataPoint<number, Date>>
+    | Array<ColumnDataPoint<Date, number>>
+    | Array<ColumnDataPoint<Date, Date>>
+    | Array<ColumnDataPoint<string, string>>
+    | Array<ColumnDataPoint<string, number>>
+    | Array<ColumnDataPoint<string, Date>>
+    | Array<ColumnDataPoint<number, string>>
+    | Array<ColumnDataPoint<Date, string>>;
+
+export interface ColumnDataPoint<X extends number | Date | string, Y extends number | Date | string> {
+    x: X;
+    y: Y;
+}
+
+export function ColumnChart({
+    name,
+    presentation,
+    series,
+    showLabels,
+    xAxisLabel,
+    yAxisLabel,
+    showLegend,
+    sortOrder,
+    style,
+    warningPrefix
+}: ColumnChartProps): ReactElement | null {
+    const dataTypesResult = useMemo(() => getDataTypes(series), [series]);
+    const sortedSeries = useMemo(() => {
+        const seriesDataType = dataTypesResult as DataTypeResult | undefined;
```

Comment:
This cast is unsafe. If the `sortSeriesDataPoints` function does not crash when the result is actually an `Error`, that is by sheer luck. Instead, I would immediately return the unsorted `series` if `dataTypesResult` is either an `Error` or `undefined`, as you cannot sort in those cases. TypeScript will automatically narrow the variable to `DataTypeResult` afterwards, so `sortSeriesDataPoints` would not need to deal with `undefined` anymore either.

Reply:
done